src/Tickets/Commerce/Admin_Tables/Orders_Table.php

```diff
@@ -698,4 +794,68 @@ protected function gateways_dropdown( $post_type ) {
 		remove_filter( 'tribe_field_div_start', '__return_empty_string', 10 );
 		remove_filter( 'tribe_field_div_end', '__return_empty_string', 10 );
 	}
+
+	/**
+	 * Displays a dropdown for filtering items in the list table by month.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $post_type The post type.
+	 *
+	 * @return void
+	 */
+	protected function post_parent_dropdown( $post_type ) {
+		/**
+		 * Filters whether to remove the 'Event' drop-down from the order list table.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool   $disable   Whether to disable the drop-down. Default false.
+		 * @param string $post_type The post type.
+		 */
+		if ( apply_filters( 'tec_tc_orders_disable_post_parent_dropdown', false, $post_type ) ) {
+			return;
+		}
+
+		// Event options are being filtered in the Frontend after the user starts typing in the search box.
+		// Except for when the user has already filtered by an event. We take the event ID from the URL and add it to the dropdown.
+
+		$e = absint( $_GET['tec_tc_events'] ?? 0 ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+
+		$event = $e ? get_post( $e ) : null;
+
+		$event = $event instanceof WP_Post ? $event : null;
+
+		$events_formatted = [
+			'' => esc_html__( 'All Events', 'event-tickets' ),
+		];
+
+		$events_formatted += $event ? [ (string) $event->ID => apply_filters( 'the_title', $event->post_title ) ] : [];
+
+		$field = [
+			'type'         => 'dropdown',
+			'options'      => $events_formatted,
+			'allow_clear'  => true,
+			'can_be_empty' => true,
+			'attributes'   => [
+				'data-freeform'              => true,
+				'data-force-search'          => true,
+				'data-searching-placeholder' => esc_attr__( 'Searching...', 'event-tickets' ),
+				'data-source'                => 'tec_tc_order_table_events',
+			],
+		];
+
+		add_filter( 'tribe_field_start', '__return_empty_string' );
+		add_filter( 'tribe_field_end', '__return_empty_string' );
+		add_filter( 'tribe_field_div_start', '__return_empty_string' );
+		add_filter( 'tribe_field_div_end', '__return_empty_string' );
+		?>
+		<label for="tec_tc_events-select" class="screen-reader-text"><?php esc_html_e( 'Filter By Event', 'event-tickets' ); ?></label>
+		<?php
+		new Tribe__Field( 'tec_tc_events', $field, $e );
+		remove_filter( 'tribe_field_start', '__return_empty_string', 10 );
+		remove_filter( 'tribe_field_end', '__return_empty_string', 10 );
+		remove_filter( 'tribe_field_div_start', '__return_empty_string', 10 );
+		remove_filter( 'tribe_field_div_end', '__return_empty_string', 10 );
```

Comment:
I would just print the field manually, using `Tribe__Field` creates more problems than it solves honestly here.